### PR TITLE
Union Pooler Capacity Experiment

### PR DESCRIPTION
**Pull request by rmccall84**

### CODE REVIEW

**Comment by rmccall84** [comment]
@scottpurdy ready for review

**Comment by rmccall84** [comment]
@scottpurdy rdy for review

File: union_pooling/union_pooling/experiments/capacity/capacity_experiment.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+from nupic.data.generators.pattern_machine import PatternMachine
+from nupic.data.generators.sequence_machine import SequenceMachine
+from nupic.research.monitor_mixin.monitor_mixin_base import MonitorMixinBase
+
+from experiments.capacity import data_utils
+from union_pooling.experiments.union_pooler_experiment import (
+    UnionPoolerExperiment)
+
+"""
```
Comment (scottpurdy) [inline union_pooling/union_pooling/experiments/capacity/capacity_experiment.py]: Module doc string goes before imports

File: union_pooling/union_pooling/experiments/capacity/capacity_experiment.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+import csv
```
Comment (scottpurdy) [inline union_pooling/union_pooling/experiments/capacity/capacity_experiment.py]: leave a blank line after copyright

File: union_pooling/union_pooling/experiments/capacity/capacity_experiment.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+import random
+import sys
+import time
+import os
```
Comment (scottpurdy) [inline union_pooling/union_pooling/experiments/capacity/capacity_experiment.py]: order alphabetically

File: union_pooling/union_pooling/experiments/capacity/capacity_experiment.py [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+import yaml
+from optparse import OptionParser
+
+import numpy
+from pylab import rcParams
```
Comment (scottpurdy) [inline union_pooling/union_pooling/experiments/capacity/capacity_experiment.py]: doesn't appear to be used

File: union_pooling/union_pooling/experiments/capacity/capacity_experiment.py [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+  return generatedSequences, labeledSequences
+
+
+def runTestPhase(experiment, inputSequences, seqLabels, sequenceCount,
+                 sequenceLength, consoleVerbosity):
+  print "\nRunning Test Phase..."
+  unionSdrs = []
+  for i in xrange(sequenceCount):
+    seq = inputSequences[
+          i + i * sequenceLength: i + 1 + (i + 1) * sequenceLength]
```
Comment (scottpurdy) [inline union_pooling/union_pooling/experiments/capacity/capacity_experiment.py]: this line is impossible to read

File: union_pooling/union_pooling/experiments/capacity/capacity_experiment.py [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+  return unionSdrs
+
+
+def run(params, paramDir, outputDir, plotVerbosity=0, consoleVerbosity=0):
+  """
+  Runs the Union Pooler capacity experiment.
+
+  :param params: A dict containing the following experiment parameters:
+
+        patternDimensionality - Dimensionality of sequence patterns
```
Comment (scottpurdy) [inline union_pooling/union_pooling/experiments/capacity/capacity_experiment.py]: nice documentation!

**Comment by rmccall84** [comment]
@scottpurdy thank you for the review. I've made your suggested changes in a new PR since I already closed these

**Pull request merged**